src/logic/map_objects/immovable_program.cc

```diff
@@ -262,7 +251,7 @@ ImmovableProgram::ActTransform::ActTransform(std::vector<std::string>& arguments
 		descr.becomes_.insert(
 		   std::make_pair(bob_ ? MapObjectType::BOB : MapObjectType::IMMOVABLE, type_name_));
 		if (!bob_) {
-			immovable_relations_.push_back(std::make_pair(descr.name(), type_name_));
+			descr.descriptions_.add_immovable_relation(descr.name(), type_name_);
```

Comment:
Instead of accessing a public member, you could add a wrapper function for the required methods.

```suggestion
			descr.add_immovable_relation(descr.name(), type_name_);
```